### DISCUSSION TITLE
announcement of turn instructions (voice)  earlier

### DISCUSSION THREAD

**Discussion by EssBee59** [osmandapp/OsmAnd#11704] [General]
Hello,

I am using osmand on my bike and often experience that the turn-by-turn instructions (voice) come too late.

"too late" means, right at the turning point or a few metres after.
I oberved that by navigating with track but also with routing.

I do not know, what is exactly the cause of that: 
- My smartphone is very fast and has "dual band" GNSS, I think this is not the cause.
- smal errors in the OSM map (some meters) could be a cause, difficult to verify (impossible to fix)  

As I am using the "display on / off" option, the screen is generally off and is reactivated by the turn instruction: not a very confortable situation when the instruction comes too late.

Would it be possible to post the "TURN NOW" some meters earlier?
(I think about a value between 5 and 10 meters, defined as a parameter?)
regards
Ess Bee

**Comment by sonora** [comment]
Here is our time table for this: http://docs.osmand.net/en/main@latest/development/algorithms/voice-prompt-triggering

Do you observe this documented behavior, or if there is what you see considerably different? Also, are your turn arrow icon and voice prompts in sync, or this there a significant discrepancy?

If it is only the voice prompts affected in your system, you may try as a workaround the Bluetooth delay compensation I had once built in: Enable the Development Plugin, then set via its settings/Test voice prompts/button 11.2 (up to 3 seconds)

**Comment by EssBee59** [comment]
Hello sonora,
Big thank for the answer and the detailed documentation.

I will check the points above and test with different parameters
(announcement time --> early  and / or chnage the default speed from 10 to 20 km/h) 

I give you feed back anyway, need some time
Ess Bee

**Comment by pebogufi** [comment]
Thanks for the link to docu.
I can confirm the problem of @EssBee59.
Same situation, cycling with screen of and reactivation by turn instruction.  
Instruction comes up at +/-  2-3m from turning point, with announcement set to "early".  According to the table it should come up at 15 to 18 m. (10 to 12 * 1.5) before. 
Maybe too much time is lost by the waking up process of the device ??
Due to waking up of screen i really can't see whether arrow icon and voice prompt are at the same time.
I also will make more tests in the next days and report.

The "turn in x m" instruction lools ok, there is enough time before event.

**Comment by EssBee59** [comment]
Again, thank a lot for your help!

I tested with the following new settings:

"Default speed"
For fastbike 25 km/h
For trekking/MTB 20 km/h

"Announcement time"  --> early

And the voice announcement IS now PERFECT!
Sorry to have disturb you, the parameters in OSMAND are fantastic!
Just perhaps a recommendation:
In the setup dialog for "Default speed", a minor change could help new users...
Replacing as example (may affect routing) with something like (may affect routing and the time of voice announcement) 

Regards
Ess Bee

**Comment by EssBee59** [comment]
And feel free to close this issue!

**Comment by pebogufi** [comment]
But what about the estimated time of arrival, i suppose, it is now worse - or not ??
Edit: may be not worse, but at least different.
Aftersome calculations, ETA is not changed by changing default speed from 10 to 25, but then - what is it good for, if it does not change ETA ?

**Comment by EssBee59** [comment]
Sorry, I never had a look at ETA...
The voice instructions are essential during my navigation with display on/off
On the other hand, 20 km/h is not unpossible?
(we could also suggest to set the default value for "default speed" for biking to 20 instead 10 as now...but it is not important for me now)
